controllers/etcdpeer_controller.go

```diff
@@ -580,10 +583,17 @@ func (r *EtcdPeerReconciler) updateStatus(peer *etcdv1alpha1.EtcdPeer, serverVer
 	peer.Status.ServerVersion = serverVersion
 }
 
+func (r *EtcdPeerReconciler) updateHealth(peer *etcdv1alpha1.EtcdPeer, healthy bool) {
+	peer.Status.Healthy = healthy
+}
+
 func (r *EtcdPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
 	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
 	defer cancel()
 	log := r.Log.WithValues("peer", req.NamespacedName)
+	if r.lastHealthy == nil {
```

Comment:
That's not gonna race ? The calls to Reconcile will always be serialised?

Reply:
Yep, fixed in #22 I'll merge it as one commit